src/fnargs.c

```diff
@@ -136,7 +159,17 @@ int prepare_fn_args_specs(int func_id, const struct mass_attacher_func_info *fin
 	struct func_arg_spec *spec;
 	const struct btf_type *fn_t, *t;
 	int i, n, reg_idx = 0;
+#if defined(__x86_64__)
 	int stack_off = 8; /* 8 bytes for return address */
+#elif defined(__aarch64__)
+	/* From the AArch64 ABI Function Call Standard: "The next stacked argument
+	 * address (NSAA) is set to the current stack-pointer value (SP)."
+	 */
+	int stack_off = 0; 
+						
+#else
+	int stack_off = 0;
+#endif
```

Comment:
can you please also adjust

```
fn_args->arg_spec_cnt = 6;
```

below to take into account architecture's maximum in register argument count? We probably need `#define` for that